### PR TITLE
Thaum Rebalance

### PR DESCRIPTION
**Pull request by Alastors**
As stated in the title, and as has been discussed in #Meta-Dev before a year or so ago, this PR focuses on rebalancing thaum to start in steam, because as has been said before, Thaumcraft functions best as early-game as possible, because unlike GT, it doesn't have an exact linear progress, and a lot of its content starts to taper off after HV, which is super unfortunate given its current tiering. 

The biggest thing this does, in combination with its sister pr ([link](https://github.com/GTNewHorizons/GTNH-TC-Wands/pull/18)) is to rebalance the wands, alongside certain other recipe tweaks here and there. 

You may ask yourself, "Alastor, but how are they going to get the shards for everything?" 
There's two solutions to this:
1. We could add shards to the OW small ore generation like silver
2. There's already multiple villagers that outright sell shards

Infusion may need to be tweaked, I aim to move it to MV, but in reality, it's a bit more like late LV, since it's still based on aluminium screwed wands to obtain.

I hardly touched any content past HV, the only one of note is the Aura Pylon, but honestly it was way over tiered for years now given how terrible the majority of its effects are.

### CODE REVIEW

**Review by Dream-Master — CHANGES_REQUESTED** [review]
This destroy the early game progession of tc wands and I am against it. You change stainless to Alubrass which mean hv to lv ? And btw I don’t allow shards in ow gen . We was clear we not touch ow Worldgen anymore .

**Comment by Alastors** [comment]
> This destroy the early game progession of tc wands and I am against it. 

That's the point, this needs to happen, thaum starts too late.

> You change stainless to Alubrass which mean hv to lv ? And btw I don’t allow shards in ow gen .

Alu brass? What are you talking about? I switched Stainless to Aluminium. It's a single tier down between LV-HV for select items and wands

> We was clear we not touch ow Worldgen anymore .

That's fine, like I said this doesn't need to change. There are already ways to get OW shards

File: src/main/java/com/dreammaster/scripts/ScriptThaumcraft.java [hunk 1]
```diff
@@ -2319,7 +2320,7 @@ private void alchemy() {
                         .add(Aspect.getAspect("sano"), 16).add(Aspect.getAspect("aqua"), 32)
                         .add(Aspect.getAspect("machina"), 8),
                 getModItem(Thaumcraft.ID, "blockJar", 1, 0, missing),
-                new ItemStack[] { CustomItemList.StainlessSteelBars.get(1L),
+                new ItemStack[] { CustomItemList.AluminiumBars.get(1L),
```
Comment (Dream-Master) [inline src/main/java/com/dreammaster/scripts/ScriptThaumcraft.java:2323]: Here

Comment (Alastors) [reply src/main/java/com/dreammaster/scripts/ScriptThaumcraft.java:2323]: That says Aluminium **_Bars_**

**Comment by Dream-Master** [comment]
When we add the new village mode I will rework trades too. And shards  not belongs there .

**Comment by Alastors** [comment]
> When we add the new village mode I will rework trades too. And shards  not belongs there .

What do you mean the shards do not belong there, that's literally the point of the thaumaturge villager, he only sells shards

**Comment by Alastors** [comment]
I mean I can add a shitty way to get shards in steam, to then switch out to the TF in LV like most other thing

**Comment by Alastors** [comment]
The only thing that changes between these two is a direct shift down a single tier of less than a quarter of items + wands, in a cohesive and sensible way as to maintain the precise balance we already have in the pack.

I'll note I don't touch the majority of offenders such as the golem cores which absolutely have no business being HV. I primarily only touch things in the main line or things that see 0 use in the pack as a whole like the item grate

**Comment by joshaup1999** [comment]
I do think it would be nice if we had thaum during steam or LV. Early game is semi void of magic which is where it is more likely to flourish than mid to late game

**Comment by Dream-Master** [comment]
> I do think it would be nice if we had thaum during steam or LV. Early game is semi void of magic which is where it is more likely to flourish than mid to late game

You can have Alu mid lv game so lv is not late game

**Comment by Alastors** [comment]
> > I do think it would be nice if we had thaum during steam or LV. Early game is semi void of magic which is where it is more likely to flourish than mid to late game
> 
> 
> 
> You can have Alu mid lv game so lv is not late game 

That's not the point, the point is thaum has a lot of moving parts that involve going places and scanning things and finding objects, all things that are very very early game, and thusly it makes more sense for it to start in steam than the arbitrary "LV" even though it officially starts at MV

**Comment by Alastors** [comment]
I will note, this is preceding some massive new content for thaumcraft's relative endgame as part of the Planar Artifice backport such as 6 additional tiers of Alchemical Furnace as well as new endgame content through its bismuth backports, or rather, new interpretations of its bismuth content in a thaum 4 lens.

**Comment by mitchej123** [comment]
+1 to spreading thaum over more tiers and making it available earlier

**Comment by chochem** [comment]
> +1 to spreading thaum over more tiers and making it available earlier

this PR is not spreading thaum over more tiers.

**Comment by Steelux8** [comment]
I can vouch for earlier Thaum, especially for less experienced players who do take a longer time to progress through the beginning of the game. Having magic as an option there should allow for more enjoyable gameplay.

**Comment by YannickMG** [comment]
I've played with people who were only really interested in how this pack handles magic progression and it was really annoying for them to have to wait all this time to get the first wand, so having it be available earlier would be nice.

**Comment by Dream-Master** [comment]
> I've played with people who were only really interested in how this pack handles magic progression and it was really annoying for them to have to wait all this time to get the first wand, so having it be available earlier would be nice.

if there is a complete written comcept for the whole new tier system we can start discussing it. Atm its nuts.

**Comment by Alastors** [comment]
> > I've played with people who were only really interested in how this pack handles magic progression and it was really annoying for them to have to wait all this time to get the first wand, so having it be available earlier would be nice.
> 
> 
> 
> if there is a complete written comcept for the whole new tier system we can start discussing it. Atm its nuts.

It's a very simple concept right now, wands, infusion, and a few things with screws shift down one tier

**Comment by Alastors** [comment]
> > +1 to spreading thaum over more tiers and making it available earlier
> 
> 
> 
> this PR is not spreading thaum over more tiers.

As I said, in combination with incoming new content it will be, and already does given the majority of current content doesn't shift.

**Comment by chochem** [comment]
if you are planning new TC content for IV-ZPM, why wouldnt you want wands there?

**Comment by Alastors** [comment]
> if you are planning new TC content for IV-ZPM, why wouldnt you want wands there?

Because it doesn't currently make sense where the ichor wand sits for the moment, it's an arbitrary last step when wand crafting doesn't matter and you can genuinely make a wand large enough to craft everything in the pack with a decent enough discount in EV, so it's logical to move the ichor one into IV which I did, this opens room for new wand cores and caps centered around Planar Artifice in the slightly later game. And perhaps even actual crafting recipes larger than even the ichor wands can craft. New wand cores starting at 2.5k vis, new caps with new animations.

There's a lot of room for improvement that ichor randomly being in the midgame doesn't serve well and will only lead to increasing annoyance from players about the increasing irrelevance of Thaumic Tinkerer as the new content comes

**Comment by Alastors** [comment]
Update on this, I'm working on a fully fledged analysis, but it's going to take a bit. Just wanted to put that out there since the freeze is coming.

**Comment by IncandescenceSy** [comment]
I'm extremely in favor of moving Thaumcraft stuff to be earlier in general; so much of the mod is just not worth using by the time you can get it, and it feels bad. Being able to start it earlier should give more options for things to do in earlygame and make players that enjoy magic more engaged

For getting thaumic shards pre-TF, perhaps you could add a (bad) Crucible recipe for turning some gem (maybe Quartzite, Diamond, or something else, idk) into a shard with some of the matching Essentia

**Comment by Dynamiczbee** [comment]
Just want to add my own support here for down-tiering Thaum's earliest stages to steam. I took so long even getting my first EBF when I first started playing GTNH, hundreds of hours over multiple worlds. I'm uniquely bad at progressing. I wish I could have accessed TC sooner as someone who'd never touched it before.

**Comment by serenibyss** [comment]
Closing since alastor is no longer around, and if this is going to be picked up again by someone else, it should start as a discussion on discord (and talking to admins) rather than a continuation of this pr on github

**Comment by Alastors** [comment]
I'll put a small comment on this as a post-mortem since the email hit me.

This PR, evidently, won't ever be accepted, whether it had been from myself, or anyone else. The internal politics have made it very clear that both Dream and Boubou, for whatever reason, are incredibly against moving thaumcraft down any tiering.

I'm not sure what advice I'd give someone who would be interested in this project. However, I will advise you that given the changes that are being made to the development cycle, as well as the intense internal politics that for some reason surround Thaumcraft, you're better off doing something else. Even if this change would've brought so much new life and potential into a mod so beloved.

**Comment by Dream-Master** [comment]
@Alastors It has nothing to do with Thaumcraft leveling up or down a tier. You can't just bluntly go through this tier change, you have to think about what new tier items you need to reach a certain level. I would have beaten anyone else over the head with it. 
Don't be so convinced and think it has something to do with you.

**Comment by Alastors** [comment]
> @Alastors It has nothing to do with Thaumcraft leveling up or down a tier. You can't just bluntly go through this tier change, you have to think about what new tier items you need to reach a certain level. I would have beaten anyone else over the head with it. 
> 
> Don't be so convinced and think it has something to do with you.

Dream, respectfully I never argued it was about me.

I can pull up the screenshots where you directly said you were against it. I am attempting to leave things vague as a push to maintain what goodwill I have for the team, and I do not appreciate this nonsense, and I intend to not waste anyone else's time with a fool's errand.

**Comment by Dream-Master** [comment]
Yes, I don't like the change you made, that's correct. But that you can talk about tier changes or even talk about when Thaumcraft will be available I have never rejected. When I started 2014/15  it took a long time to integrate all mods nicely and I don't want that to be destroyed by a quick PR on your part.

**Pull request closed without merging**